File: eenass/QLanguage/src/ast/types/BoolType.java
package ast.types;

import ast.visitors.TypeVisitor;

public class BoolType extends Type {

	@Override
	public <T> T accept(TypeVisitor<T> visitor) {
		return visitor.visit(this);
	}

	@Override
	public boolean isCompatibleTo(Type type) {
		return (type.isCompatibleToBool() && !type.isCompatibleToInt() && !type.isCompatibleToStr());
	}
	
	@Override
	public boolean isCompatibleToBool(){
		return true;
	}

}

File: rperz/QSLib/QSLib/AST/Types/BoolType.cs
using System.Windows;
using System.Windows.Controls;
using System;
using QSLib.GUI;
using QSLib.Visitors;
namespace QSLib.AST.Types
{
    public class BoolType : QSType
    {
        public override string Name
        {
            get { return "Boolean"; }
        }

        public override bool IsCompatible(QSType other)
        {
            if (other == null)
                return false;
            return other.IsBoolean();
        }

        public override object Accept<T>(ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override bool IsBoolean()
        {
            return true;
        }

        public override string ToString()
        {
            return "boolean";
        }

        public override object GetUndefined()
        {
            return false;
        }

    }
}
